### DISCUSSION TITLE
How to get return when I use web3Modal.open()?

### DISCUSSION THREAD

**Discussion by 0xNullMe** [WalletConnect/walletconnect-monorepo#3314] [Developer Support] [answered]
when i try to use the https://github.com/WalletConnect/web3modal-examples/tree/main/web3modal-wagmi-html-cdn, i has been read wagmi and web3modal doc.Still don't understand how to get return. What should I do if I refuse to connect or if the connection is successful?

**Comment by glitch-txs** [comment]
Once the user connects you can use wagmi actions to interact with the wallet
https://wagmi.sh/core/getting-started

**Reply by glitch-txs** [reply, marked as answer]
You can subscribe to the modal and get the account when closed
https://docs.walletconnect.com/2.0/web/web3modal/html/wagmi/actions#subscribemodal
You can also use the watchAccount subscriber from wagmi
